src/main/java/org/kohsuke/github/GHBranchProtection.java

```diff
@@ -221,6 +221,54 @@ public boolean isEnabled() {
         }
     }
 
+    /**
+     * The type Check.
+     */
+    public static class Check {
+        @JsonProperty
+        private String context;
+
+        @JsonProperty
+        private Integer app_id;
+
+        /**
+         * no-arg constructor for the serializer
+         */
+        public Check() {
+        }
+
+        /**
+         * Regular constructor for use in user business logic
+         *
+         * @param context
+         *            the context string of the check
+         * @param appId
+         *            the application ID the check is supposed to come from
+         */
+        public Check(String context, Integer appId) {
+            this.context = context;
+            this.app_id = appId;
+        }
+
```

Comment:
```suggestion
```

Reply:
If this has no constructor, how can the user use the `GHBranchProtectionBuilder.addRequiredChecksWithAppIds` method, which accepts this object?

Reply:
Jackson is magic.  It will do the binding for you.

Reply:
I don't think Jackson is related to the use case I am referring to. Here is an example of the use-case:

https://github.com/hub4j/github-api/blob/3882581deaedb92366c5a7b62b69a26c41f1113e/src/test/java/org/kohsuke/github/GHBranchProtectionTest.java#L203-L204

Or do you mean the user should use json strings and then Jackson to instantiate objects to pass in as arguments to that method?

Reply:
Ah, I see.  My mistake.